File: 2labz/2labz2/2.rb
def normalize_email(email)
  email = email.strip.downcase  # Удаляем пробелы и приводим к нижнему регистру
  username, domain = email.split('@')
  username = username.gsub('.', '').split('+').first
  "#{username}@#{domain}"
rescue
  nil # Возвращаем nil для некорректных email-адресов
end

puts "Введите email-адреса через запятую:"
input = gets.chomp
emails = input.split(',').map(&:strip)  # Разделяем по запятым и убираем пробелы

# Нормализуем и удаляем дубликаты и некорректные адреса
normalized_emails = emails.map { |email| normalize_email(email) }.compact.uniq

puts "\nУникальные адреса после нормализации:"
normalized_emails.each { |email| puts "- #{email}" }

puts "\nКоличество уникальных адресов: #{normalized_emails.size}"

File: 2labz/2labz2/2.py
def normalize_email(email):
    email = email.strip()  # Удаляем пробелы в начале и конце
    username, domain = email.split('@')
    username = username.replace('.', '')          # Удаляем все точки
    username = username.split('+')[0]             # Игнорируем часть после +
    return f"{username.lower()}@{domain.lower()}"  # Приводим к нижнему регистру

print("Введите email-адреса через запятую:")
input_emails = [email.strip() for email in input().split(',')]  # Разделяем по запятым и убираем пробелы

normalized_emails = []
for email in input_emails:
    try:
        normalized = normalize_email(email)
        normalized_emails.append(normalized)
    except ValueError:
        print(f"Некорректный email: {email}")

unique_emails = set(normalized_emails)           # Уникальные адреса

print("\nУникальные адреса после нормализации:")
for email in sorted(unique_emails):
    print(f"- {email}")

print(f"\nКоличество уникальных адресов: {len(unique_emails)}")